opsdroid/loader.py

```diff
@@ -29,30 +30,37 @@ def __init__(self, opsdroid):
         _LOGGER.debug("Loaded loader")
 
     @staticmethod
+    def import_module_from_spec(module_spec):
+        """Import from a given module spec and return imported module."""
+        module = importlib.util.module_from_spec(module_spec)
+        module_spec.loader.exec_module(module)
+        return module
+
+    @staticmethod
     def import_module(config):
         """Import module namespace as variable and return it."""
-        try:
-            module = importlib.import_module(
-                config["module_path"] + "." + config["name"])
-            _LOGGER.debug("Loaded %s: %s", config["type"],
-                          config["module_path"])
-            return module
-
-        except ImportError as error:
-            _LOGGER.debug("Failed to load %s.%s.  ERROR: %s",
-                          config["module_path"], config["name"], error)
-
-        try:
-            module = importlib.import_module(
-                config["module_path"])
-            _LOGGER.debug("Loaded %s: %s", config["type"],
-                          config["module_path"])
+        # Check if the module can be imported and proceed with import
+
+        # Proceed only if config.name is specified
+        # and parent module can be imported
+        if config["name"] and importlib.util.find_spec(config["module_path"]):
+            module_spec = importlib.util.find_spec(config["module_path"] +
+                                                   "." + config["name"])
+            if module_spec:
+                module = Loader.import_module_from_spec(module_spec)
+                _LOGGER.debug("Loaded " + config["type"] +
+                              ": " + config["module_path"])
+                return module
+
+        module_spec = importlib.util.find_spec(config["module_path"])
+        if module_spec:
+            module = Loader.import_module_from_spec(module_spec)
+            _LOGGER.debug("Loaded " + config["type"] +
```

Comment:
We have recently changed the logging format to use % instead of concatenation. I might be wrong but you might want to change the logging to use %s (like in the code that was replaced)

example: `_LOGGER.error("Failed to load %s: %s", config["type"], config["module_path"])`
If you need any help just let me know 😃

Reply:
Sure @FabioRosado. Even I was unsure which format to follow. I'll ensure my changes will have % format. Thanks